### ISSUE TITLE
`uni.Dict()` extremely slow on iOS with CGO

### ISSUE THREAD

**Issue by dariyooo** [ikawaha/kagome-dict#34]
Hello, thank you for this nice package again!

I am currently trying to build bindings for this package (kagome with uniDict) for dart. This works great on Windows, MacOS and Android. However, I am having some problems on iOS.

I am compiling this library using CGO and then bundling it into a xcframework ([here](https://github.com/CaptainDario/kagome_dart/blob/2de800f10d92ba97d1d592aafcfb3dc106adce76/Makefile#L112) is the makefile). While this works, instantiating an instance of iPadic is extremely slow on my iPad Pro 2017. It takes smth between 5-10 minutes.
Specifically, I am creating a global Tokenizer and with an ipadDict instance [code](https://github.com/CaptainDario/kagome_dart/blob/2de800f10d92ba97d1d592aafcfb3dc106adce76/kagome.go#L46).

``` go
// global kagome tokenizer
var globalTokenizer tokenizer.Tokenizer

...

//export InitTokenizer
func InitTokenizer() {
	
	tokenizer, err := tokenizer.New(uni.Dict(), tokenizer.OmitBosEos())
	if err != nil {
		panic(err)
	}

	globalTokenizer = *tokenizer;
	globalTokenizerInitialized = true;
}
```

The weird thing is that on all other platforms it is really fast (< 2s). Do you have any idea what could be the problem?

**Comment by ikawaha** [comment]
It seems to be an system dependent issue. The reason for this I do not know. One thing I would try is to use the previous version v1.0.5, as I have changed the way the dictionaries are embedded since v1.0.6.

Thank you for the feedback.

**Comment by dariyooo** [comment]
Are you sure v1.0.5? The changelog of 1.0.5 says that there you introduced the change?

**Comment by dariyooo** [comment]
I tested with kagome-dict@v1.0.4, this is the go.mod file:

```
module example.com/m/v2

go 1.19

require (
	github.com/ikawaha/kagome-dict/uni v1.1.3
	github.com/ikawaha/kagome/v2 v2.8.0
)

require github.com/ikawaha/kagome-dict v1.0.4 // indirect
```

There is no significant change in loading time.
However, when I store the dictionary instance as a global variable the loading time is less (< 50s).

```go
var globalDict = uni.Dict();

...

//export InitTokenizer
func InitTokenizer() {

	tokenizer, err := tokenizer.New(globalDict, tokenizer.OmitBosEos())
	if err != nil {
		panic(err)
	}
	globalTokenizer = *tokenizer;
	globalTokenizerInitialized = true;
}
```

When trying to use neologd Dict this error appears:

```
* thread #2, stop reason = EXC_RESOURCE RESOURCE_TYPE_MEMORY (limit=3072 MB, unused=0x0)
    frame #0: 0x0000000104c75a04 Runner`runtime.mallocgc + 852
Runner`runtime.mallocgc:
->  0x104c75a04 <+852>: stp    xzr, xzr, [x11]
    0x104c75a08 <+856>: ldr    x12, [x8, #0x18]
    0x104c75a0c <+860>: cmp    x0, x12
    0x104c75a10 <+864>: b.lo   0x104c75a1c               ; <+876>
Target 0: (Runner) stopped.
```
Seems to me like memory allocation is taking so long...

**Comment by dariyooo** [comment]
Do you have any other idea? Is it related to how CGO works and it is maybe worth it opening an issue there?

**Comment by ikawaha** [comment]
I am not sure if this is a good method, but there is a way to read and use dictionaries directly from a file. Is it possible to read files from the file system in this environment?

The dictionary file is here: https://github.com/ikawaha/kagome-dict/blob/main/uni/uni.dict

I hope this will work for you.

```Go
package main

import (
	"fmt"

	"github.com/ikawaha/kagome-dict/dict"
	"github.com/ikawaha/kagome/v2/tokenizer"
)

func main() {
	dict, err := dict.LoadDictFile("uni.dict") // ★
	if err != nil {
		panic(err)
	}
	t, err := tokenizer.New(dict, tokenizer.OmitBosEos())
	if err != nil {
		panic(err)
	}
	// wakati
	fmt.Println("---wakati---")
	seg := t.Wakati("すもももももももものうち")
	fmt.Println(seg)
}
```

**Comment by dariyooo** [comment]
Yes, there is a way to load files from path in Flutter. Maybe this is in my case even the better solution. Because than one does not need to recompile the library when one wants to change the dictionary.

One question, what size of RAM is recommend for using the ipa-neologd dictionary?

Thank you for your help. I am gonna try this out and report back.

**Comment by dariyooo** [comment]
@ikawaha where do I get the "uni.dict" file from? Can I use the standard iPa / uni dict file like in me cab?

**Comment by ikawaha** [comment]
Hi,

|dict|path|
|:--|:--|
| IPA | https://github.com/ikawaha/kagome-dict/blob/main/ipa/ipa.dict |
| Uni | https://github.com/ikawaha/kagome-dict/blob/main/uni/uni.dict |

Kagome is not compatible with mecab dictionary binaries.

**Comment by KEINOS** [comment]
@CaptainDario 

You may have to isolate whether it is a specification on the iOS side or a memory limitation. It might be a good idea to create a custom dictionary with several terms and test it.

**Comment by dariyooo** [comment]
@KEINOS I do not know how to create a smaller dictionary, is there one available to download?

**Comment by KEINOS** [comment]
@CaptainDario 

> I do not know how to create a smaller dictionary, is there one available to download?

Unfortunately, I don't think there is anything available for download in that size.

However, tests of `dict.go` showed that the following function generates and returns a `dict.Dict` object with three keywords. I think this function is a good start.

https://github.com/ikawaha/kagome-dict/blob/018172dab50f3d4ad7721c942e999051898865b3/dict/dict_test.go#L11-L73
